### PR TITLE
ACMS-1481: PHPUnit testcase coverage for API Menu Items.

### PR DESCRIPTION
**Pull request by rajeshreeputra**
**Motivation**
Fixes #ACMS-1481

**Proposed changes**
PHPUnit testcase coverage for API Menu Items.

**Alternatives considered**
NA

**Testing steps**

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

File: modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php [hunk 1]
```diff
@@ -98,4 +139,13 @@ protected function installModule(string $module) {
     return $this->moduleInstaller->install([$module]);
   }
 
+  protected function childMenuTest($menu, $parent, $childs, $selector) {
```
Comment (deepakmishra2) [inline modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php:154]: @rajeshreeputra Add return type for this method.

File: modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php [hunk 2]
```diff
@@ -98,4 +151,13 @@ protected function installModule(string $module) {
...
+    $this->assertEquals($parent, $menu->getText());
+    $menu->mouseOver();
+    foreach ($childs as $key => $child) {
+      $key += 1;
```
Comment (apathak18) [inline modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php:158]: @rajeshreeputra can we do $key++;

File: modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php (hunk 1 above)
Comment (apathak18) [reply modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php:154]: +1

File: modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php [hunk 3]
```diff
@@ -98,4 +151,13 @@ protected function installModule(string $module) {
     return $this->moduleInstaller->install([$module]);
   }
 
+  protected function childMenuTest($menu, $parent, $childs, $selector) {
```
Comment (apathak18) [inline modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php:154]: @rajeshreeputra add small functional description of this method.

File: modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php (hunk 2 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php:170]: done

File: modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php (hunk 3 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php]: done

File: modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php (hunk 1 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php]: done

File: modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php [hunk 4]
```diff
@@ -68,6 +76,51 @@ public function testAcmsWizardMenu(): void {
...
+      // System administration Menu.
+      $systemAdministrationMenu = $assertSession->waitForElementVisible('css', '.toolbar-icon-admin-cms');
+      $systemAdministrationMenuLi = [
+        'Structure',
+        'Extend',
+        'Configuration',
+        'People',
+        'Reports',
+      ];
+      $this->childMenuTest($systemAdministrationMenu, 'System administration', $systemAdministrationMenuLi, '.toolbar-icon-admin-cms');
```
Comment (chandan-singh7929) [inline modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php:122]: @rajeshreeputra - can we data provider for each menu and pass argument as expected child to reduce the duplicity of code?

File: modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php [hunk 5]
```diff
@@ -68,6 +76,51 @@ public function testAcmsWizardMenu(): void {
...
+      // Content Menu.
+      $contentMenu = $assertSession->waitForElementVisible('css', '.toolbar-icon-system-admin-content');
+      $contentMenuLi = [
+        'Scheduled Content',
+        'Add content',
+        'Files',
+        'Media',
+        'Scheduled Media',
+      ];
+      $this->childMenuTest($contentMenu, 'Content', $contentMenuLi, '.toolbar-icon-system-admin-content');
```
Comment (deepakmishra2) [inline modules/acquia_cms_headless/tests/src/Functional/HeadlessHybridModeTest.php:89]: Test here is failing
Screenshot 2023-02-20 at 5 19 31 PM

**Comment by chandan-singh7929** [comment]
@rajeshreeputra - Please check https://github.com/acquia/acquia_cms/actions/runs/4229418209/jobs/7346333614#step:9:205

**Review by deepakmishra2 — APPROVED** [review]
LGTM

**Review by chandan-singh7929 — APPROVED** [review]

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**